src/js/25-trending-topics-fallback.js

```diff
@@ -0,0 +1,47 @@
+;(() => {
+  'use strict'
+
+  if (!document.querySelector('atomic-recs-interface')) return
+
+  const addFallbackLinks = (parentDiv) => {
+    const trendingTopicsFallback = 'trendingTopicsFallback'
+    const title = document.createElement('h2')
+    title.innerText = trendingTopicsFallback[document.documentElement.lang].title
+    parentDiv.appendChild(title)
+    trendingTopicsFallback[document.documentElement.lang].links.forEach((link) => {
+      const div = document.createElement('div')
+      const a = document.createElement('a')
+      a.innerText = link.name
+      a.setAttribute('href', link.url)
+      a.classList.add('trending-topics-hardcoded')
+      parentDiv.appendChild(div)
+      div.appendChild(a)
+    })
+  }
+
+  const hasValidRecommendations = (recsListShadowRoot) => {
+    if (recsListShadowRoot.childElementCount === 0) return false
+    const atomicRecsResult = recsListShadowRoot.querySelector('atomic-recs-result')
+    if (atomicRecsResult) {
+      const recsResultShadowRoot = atomicRecsResult.shadowRoot
+      const atomicText = recsResultShadowRoot.querySelector('atomic-text')
+      return atomicText?.value !== 'no-title'
+    }
+    return true
+  }
+
+  setTimeout(() => {
```

Comment:
I notice that the coveo examples use something like `await customElements.whenDefined('atomic-search-interface')` to listen for the coveo elements loading. I wonder if that would work? If so, I think it would really improve the experience, as a four second content swap feels quite long.

Reply:
let me try this

Reply:
So technically, this works (as in it doesn't generate an error in the console), but it doesn't do what is intended in this case. That's because atomic-search-interface is always defined quickly when the page loads, but it takes a little bit of time to generate the shadowRoot content which is what I want to check.

I set it to 4 seconds initially in case the internet is slow, but I think 1 second may be ok.